src/js/models/menubar.js

```diff
@@ -253,6 +253,16 @@ define(function (require, exports, module) {
                         return layer !== ancestor && document.layers.selected.contains(ancestor);
                     });
                 })),
+            "no-nesting-except-artboards":
+                (document !== null) &&
+                !document.unsupported &&
+                (document.layers !== null) &&
+                !(document.layers.selected.some(function (layer) {
+                    return document.layers.ancestors(layer).some(function (ancestor) {
```

Comment:
We already had `hasSelectedAncestors`, but in this case I needed to ignore artboards in those selected ancestors. I was trying to avoid adding more cached properties like `hasSelectedAncestorsButNoArtboards`... but in this case I'm trading performance for code size, which is not a good trade.

Will revisit.